### PR TITLE
fixes #7275 - remove welcome page and replace with docs button

### PR DESCRIPTION
**Pull request by stbenjam**
I think the info could be corrected if you want to keep it, but I think it makes sense to just show the dashboard already. And:
1. I'm tried of staring at this thing :smile: 
2. Vast, vast majority of foreman users never see this page with the Foreman host registered automatically
3. Users who do see it, are using a taxonomy and see it after they've been using Foreman for a while (i.e. creating a new taxonomy for the first time)
4. The page is full of incorrect information, encouraging the user to make decisions that were already made for them in most cases ("you need a smart proxy, go install one on this machine"), or encouraging them to use features that have not worked in ages (puppet docs browser)

### CODE REVIEW

**Comment by stbenjam** [comment]
[test]

**Comment by ohadlevy** [comment]
:+1: It would be very nice to implement https://dl.dropboxusercontent.com/u/5892944/Merged-Hosts-2015-07-09.pdf page number 3

**Comment by stbenjam** [comment]
I see the intent of being helpful to the user, but the wireframe as given is misleading.  "Satellite currently does not have any hosts created."  It'd only be scoped to the current org/location and the users permissions, so there may be thousands of hosts created.

Maybe you could have a full-length welcome widget at the top of the page introducing a user to Foreman or something, which they can close themselves? I dunno.

This page has been problematic for a while, for downstream branding, and the information we're showing to upstream users.  I was hoping this would be a quick merge; not sure I want to get mixed up in host unification tasks.  I don't think my design skills are good enough anyway to implement the wireframe, so if just the deletion isn't acceptable, we can probably close this.

**Comment by stbenjam** [comment]
Alright, I thought about it, what about something like this? We still remove the welcome page, but replace the status widget with a message similar to the one in the wireframe when there's no hosts.

My sad HTML and CSS needs help probably, but anyway looks like this:

widget

I think there's a lot of good arguments for removing this welcome page, beyond just incorrectness.  What if someone wants to use only containers or only Katello content mirroring? They should be able to see the dashboard too for non-host stuff.

**Comment by stbenjam** [comment]
Just the widget

widget

File: app/helpers/dashboard_helper.rb [hunk 1]
```diff
@@ -117,4 +117,19 @@ def report_color
       :disabled_hosts => "#92A8CD"
     }
   end
+
+  def welcome_links
+    [
+      [_("Create Host"), hash_for_new_host_path, :class => 'btn btn-success'],
+      [_("Discovered Hosts"), hash_for_new_host_path, :class => 'btn btn-default']
```
Comment (stbenjam) [inline app/helpers/dashboard_helper.rb]: This would get moved to the plugin, and discovery can extend this method with alias_method_chain.

**Comment by ehelms** [comment]
One idea that had crossed my mind while migrating dashboard components was to make the welcome/get started screen at / only and the /dashboard always be the dashboard.

**Comment by stbenjam** [comment]
>  One idea that had crossed my mind while migrating dashboard components was to make the welcome/get started screen at / only and the /dashboard always be the dashboard.

What's the argument for keeping it anywhere? I'm pretty firmly in the "kill it with fire" camp for `dashboard/welcome.html.erb`, :fire: :skull:  It's rarely seen anyway.

**Comment by domcleal** [comment]
The widget suggestion looks useless to me, let's not have that.

I'm fine with removing the text, but perhaps a better way would be to keep it on a separate Help page linked either from the dashboard or About (as it currently is) and to refresh the text.

**Comment by stbenjam** [comment]
> I'm fine with removing the text, but perhaps a better way would be to keep it on a separate Help page linked either from the dashboard or About (as it currently is) and to refresh the text.

How about a documentation button next to the Manage dashboard - that goes to the web site?

**Comment by domcleal** [comment]
Suits me, or even just put it in that menu so it's not taking up screen estate.

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -393,10 +393,15 @@ def show_parent?(obj)
     (obj.new_record? && obj.class.count > 0) || (!obj.new_record? && obj.class.count > 1)
   end
 
-  def documentation_button(section)
+  def documentation_button(section = nil)
+    url = if section
+            "http://www.theforeman.org/manuals/#{SETTINGS[:version].short}/index.html##{section}"
+          else
+            "http://www.theforeman.org/documentation.html##{SETTINGS[:version].short}"
```
Comment (stbenjam) [inline app/helpers/application_helper.rb:399]: ~~Needs https://github.com/theforeman/theforeman.org/pull/403~~

**Comment by stbenjam** [comment]
Back to my original PR + a button.  Maybe someone will click on it.

button

Yes/No?

File: app/helpers/dashboard_helper.rb [hunk 3]
```diff
@@ -16,7 +16,9 @@ def dashboard_actions
        content_tag(:li, '', :class=>'widget-add') do
          widgets_to_add
        end
-     )]
+     ),
+     documentation_button
```
Comment (domcleal) [inline app/helpers/dashboard_helper.rb]: It's not your fault, but the indentation's all off here, it's an odd number of spaces.  If you wouldn't mind fixing it, that'd be appreciated.

**Comment by domcleal** [comment]
Fine by me - it's wrapping for me, but that's just the narrow width of my monitor, which I guess isn't typical usage.

screenshot from 2015-07-29 14 13 54

Having the docs link prominently displayed is probably a good idea.

A couple of other observations:
1. The first use instructions link on the About page is still there, but broken.
2. Your screenshot is missing a search bar - is that unrelated, or because there are no hosts?

**Comment by stbenjam** [comment]
> Fine by me - it's wrapping for me, but that's just the narrow width of my monitor, which I guess isn't typical usage.

I could rename "Manage Dashboard" to just "Dashboard," that would allow slightly smaller windows.

> Your screenshot is missing a search bar - is that unrelated, or because there are no hosts?

No hosts in that org, its there when there's hosts.

**Comment by domcleal** [comment]
> > Fine by me - it's wrapping for me, but that's just the narrow width of my monitor, which I guess isn't typical usage.
> 
> I could rename "Manage Dashboard" to just "Dashboard," that would allow slightly smaller windows.

"Manage" might be better, since it describes what the menu's for?  Either way, that'd be good, thanks.

> > Your screenshot is missing a search bar - is that unrelated, or because there are no hosts?
> 
> No hosts in that org, its there when there's hosts.

Hm yeah, I suspected so, seems it's part of the `welcome` filter in app controller.  I'd suggest that the search bar should always be displayed for consistency - I realise it's not going to do much with no hosts, but having it appear and disappear is unexpected.

I suspect the code's intended only for pages where the welcome view's being rendered.

**Comment by stbenjam** [comment]
Updated, searchbar always shows on dashboard now

File: app/helpers/application_helper.rb [hunk 4]
```diff
@@ -200,8 +200,7 @@ def locked_icon(condition, hovertext)
   end
 
   def searchable?
-    return false if (SETTINGS[:login] and !User.current )
-    return false unless @searchbar
+    return false if (SETTINGS[:login] and !User.current) || @welcome
```
Comment (domcleal) [inline app/helpers/application_helper.rb]: Nitpick, change `and` to `&&` so the operators are consistent.

**Comment by domcleal** [comment]
Merged as 313979b, thanks @stbenjam.

**Pull request closed without merging**